### PR TITLE
Multiple genome catalogues

### PR DESCRIPTION
**Pull request by SandyRogers**
This PR:
- adds support for multiple genome catalogues for multiple biomes.
  - replaces `Release` model with a `GenomeCatalogue` model. 
  - replaces Many2Many relationship fro `Genome`s to `GenomesCatalogue`s with a One2One
    - this is because when catalogues are made, a genome that is identical to one from another catalogues(/biome) would be newly accessioned anyway. We may in future link them, but are unlikely to ever dereplicate them. 
    - we agreed to only support the **single latest version** of a catalogue on the website. This means before a new version is added, the previous one will be deleted including all of its genomes. This is done via django admin panel, no special management command.
  - adds some metadata fields for `GenomeCatalogue`s, because things like biome and description were previously hardcoded in frontends, assuming we only had a gut catalogue.
  - the db migration also includes some data migration help for the UHGG 1.0 catalogue, so that this can be released whilst still supporting that first ~Release~ Catalogue.
- fixes a few inconsistencies in `Download`s, and adds an env-var configurable option for downloads to be served directly, instead of by Nginx. This is just helpful for local development.

### CODE REVIEW

File: emgapi/views.py [hunk 1]
```diff
@@ -1165,6 +1161,57 @@ def list(self, request, *args, **kwargs):
...
+                             emg_viewsets.BaseGenomeCatalogueGenericViewSet):
+
+    filter_class = emg_filters.GenomeCatalogueFilter
+
+    lookup_field = 'catalogue_id'
+    lookup_value_regex = '[^/]+'
+
+    queryset = emg_models.GenomeCatalogue.objects.all()
+
+    def retrieve(self, request, *args, **kwargs):
```
Comment (gustavo-salazar) [inline emgapi/views.py:1175]: Why are you overwriting these methods to only call the `super` implementation? ISn't this something that would happen just by inheritance. Is it for the documentation? Or Am I missing something here?

Comment (SandyRogers) [reply emgapi/views.py:1175]: It is purely to add the doc strings which then appear in the swagger documentation – this is the pattern used currently in the codebase to achieve this, although there are other patterns we could adopt globally instead...

Comment (gustavo-salazar) [reply emgapi/views.py:1175]: Cool. Happy with that pattern, I was just curious.

**Review by gustavo-salazar — APPROVED** [review]
🌌 Happy with these changes! Good job!
I'll check the webkit now

File: emgapi/views.py [hunk 2]
```diff
@@ -1254,143 +1300,108 @@ def list(self, request, *args, **kwargs):
...
+    lookup_field = 'name'
+    lookup_value_regex = '[^/]+'
+
+    def get_queryset(self):
+        genome_set = emg_models.GenomeSet.objects.all()
+
+        if self.kwargs.get('name'):
+            genome_set = genome_set.filter(name=self.kwargs['name'])
+
+        genome_set.annotate(genome_count=Count('genome'))
```
Comment (mberacochea) [inline emgapi/views.py:1339]: Would it make sense to keep a field in the catalogue for the genomes count?. I would be more performant that having to do a count each time.

Comment (SandyRogers) [reply emgapi/views.py:1339]: definitely.

File: emgapi/views_relations.py [hunk 3]
```diff
@@ -547,33 +547,6 @@ def list(self, request, *args, **kwargs):
             .list(request, *args, **kwargs)
 
 
-# class PipelineStudyRelationshipViewSet(emg_mixins.ListModelMixin,
```
Comment (mberacochea) [inline emgapi/views_relations.py:550]: :feelsgood:

File: tests/api/test_genomes.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+@pytest.mark.django_db
+class TestGenomesAPI:
+    def test_catalogue_list(self, client, genome_catalogue):
+        url = reverse('emgapi_v1:genome-catalogues-list')
+        response = client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+        rsp = response.json()
+        assert len(rsp['data']) == 1
+        assert rsp['data'][0]['id'] == 'mandalor-1-0'
+        assert rsp['data'][0]['attributes']['name'] == 'Mandalorian Genomes v1.0'
```
Comment (mberacochea) [inline tests/api/test_genomes.py:37]: This is the way

**Review by mberacochea — APPROVED** [review]
:tophat: Sandy, brilliant!

File: tests/api/test_genomes.py (hunk 4 above)
Comment (gustavo-salazar) [reply tests/api/test_genomes.py:37]: baby yoday

**Comment by gustavo-salazar** [comment]
Hey @SandyRogers @mberacochea , just thought of mentioning this here.

Leaving the new setting `downloads_bypass_nginx: True` when running the tests locally, makes some of them to fail. I don't think this is necessarily an issue, but something to be aware when running tests. Also this might be just my setup. 🤷🏽

**Comment by SandyRogers** [comment]
> Hey @SandyRogers @mberacochea , just thought of mentioning this here.
> 
> Leaving the new setting `downloads_bypass_nginx: True` when running the tests locally, makes some of them to fail. I don't think this is necessarily an issue, but something to be aware when running tests. Also this might be just my setup. 🤷🏽

Ah good to know. I am using a different setup and config for unit tests than for regular local dev so didn't notice, sorry.

**Pull request merged**